state/factory/statedb.go

```diff
@@ -352,6 +361,9 @@ func (sdb *stateDB) PutBlock(ctx context.Context, blk *block.Block) error {
 		)
 	}
 
+	if blk.Height() == sdb.stopHeight {
```

Comment:
It's strange to add `stopHeight` inside `stateDB` component. The purpose could be achieved by refactoring the `PutBlock`